Add Django models

krysal opened the pull request:
## Description
Add the Django models according to discussion.

## Technical details
- `Case` and `Scholarship` models inherit from a `LegalResource` abstract base class.
- No single models for License or Contributor for now, their fields are embed into the previous abstrac class because are common to both resources (cases and scholarships)
- Use a single `name` field of contributor instead of `first_name` and `last_name` _for better support of different naming conventions and preferences_ ([ref](https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/))
- Added [django_countries](https://github.com/SmileyChris/django-countries) package to manage countries
- Added a `FAQ` model for Frequently Asked Questions

There are still some fields that need to be clarified if are actually used or not, like `issue, blurb, original_blurb` from cases, that appear in the [cases.csv](https://github.com/creativecommons/caselaw/blob/master/jekyll/_data/cases.csv) file, if these fields are used just to keep internal notes I added the `notes` field for that purpose.

## Checklist
<!-- Replace  the [ ] with [x] to check the boxes. --> 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin
<!-- You must read and understand the following attestation. -->

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

krysal commented:
Annex generated diagram:
![cc_legal_db_diagram](https://user-images.githubusercontent.com/9145885/83697352-c679bd80-a5cc-11ea-9c3d-0a5155d82535.png)

Links I checked:
- [Caselaw - Primary Data Google Sheet](https://docs.google.com/spreadsheets/d/1Z9IcBgdDYoeZw0Xx573ZMp5JcJDCNhOUHrj0guU9byo/edit#)
- [Cases form](https://docs.google.com/forms/d/1Ne1sh09XDZP7yVAOz3nvlylnhECaHn84fJ00pp0PwiI/viewform)
- [Scholarships form](https://docs.google.com/forms/d/e/1FAIpQLSfxxk5FWZCl3QURJqF42-FtMoWrwj1PMsdOyk2hUayU7FPB7w/viewform)
- Jekyll's [cases.csv](https://github.com/creativecommons/caselaw/blob/master/jekyll/_data/cases.csv) file
- Jekyll's [scholarships](https://github.com/creativecommons/caselaw/blob/master/jekyll/_data/scholarship.csv) file

kgodey commented on legal_db/models.py:
```diff
@@ -1,11 +1,71 @@
+class FAQ(models.Model):
+    question = models.CharField(max_length=250)
+    answer = models.TextField(blank=True, null=True)
+    created_at = models.DateTimeField(auto_now_add=True)
```
I would recommend creating another abstract model that just has created at and updated at fields, you can inherit all models from it (I usually call it `BaseModel` or something like that)

kgodey commented on legal_db/models.py:
```diff
@@ -1,11 +1,71 @@
-    title = models.CharField(max_length=50)
-    version = models.CharField(max_length=25, blank=True, null=True)
+class LegalResource(models.Model):
+    license = models.CharField(max_length=25, blank=True, null=True)
```
do we need three versions for licenses (license, version, is_ported)? we discussed a single string field.

I'll defer to @TimidRobot here.

kgodey commented on legal_db/models.py:
```diff
@@ -1,11 +1,71 @@
+    is_ported = models.BooleanField(blank=True, null=True)
+    contributor_name = models.CharField(max_length=120)
+    contributor_email = models.EmailField()
+    contribution_credit = models.BooleanField()
```
Could we make this field name a bit clearer so that we know this refers to whether the contributor wants to be credited?

krysal replied:
About `contriution_credit`? I thought of `give_contribution_credit` but it seems a bit long to me, I'll go with `give_credit`, is it better this way?

kgodey replied:
it's okay if it's long if that makes it clearer. how about `contribution_credit_requested`?

krysal replied:
Sound good too! :)

krysal replied:
Sounds good too! :)

kgodey commented on legal_db/models.py:
```diff
@@ -1,11 +1,71 @@
+    contributor_name = models.CharField(max_length=120)
+    contributor_email = models.EmailField()
+    contribution_credit = models.BooleanField()
+    link = models.URLField(max_length=2000)
```
Could we make this field name a bit clearer as well so we know what the link is supposed to be for?

kgodey commented on legal_db/models.py:
```diff
@@ -1,11 +1,71 @@
+    country = CountryField()
+    summary = models.TextField(blank=True, null=True)
     notes = models.TextField(blank=True, null=True)
+    publish = models.BooleanField(default=False)
```
I think this field name is a bit confusing too. Maybe we can call it status and have it be a ChoiceField instead of a boolean? we can have multiple statuses (unreviewed, review in progress, published, rejected), something like that.

I'll defer to @TimidRobot here too.

krysal replied:
Added as suggested.

kgodey commented on legal_db/models.py:
```diff
@@ -1,11 +1,71 @@
+    case_no = models.CharField(max_length=50, blank=True, null=True)
+    jurisdiction = models.CharField(max_length=250, blank=True, null=True)
+    court_name = models.CharField(max_length=250)
+    decision_district_city_circuit = models.CharField(
```
Maybe we can just call this decision_locality?

krysal replied:
Exactly that! I was looking for a name like this 😅

kgodey commented on legal_db/models.py:
```diff
@@ -1,11 +1,71 @@
+    en_translation_link = models.URLField(max_length=2000, blank=True, null=True)
+    cc_involvement = models.CharField(max_length=100, blank=True, null=True)
+    cc_implication = models.TextField(blank=True, null=True)
+    work_type = models.CharField(max_length=100, blank=True, null=True)
```
This seems a little unclear too, maybe add help text about what it should be?

Adding help text to all the fields is probably a good idea.

kgodey commented on legal_db/models.py:
```diff
@@ -1,11 +1,71 @@
+        return "(Scholarship) " + self.title
+
+
+class FAQ(models.Model):
```
We probably want FAQs to be ordered manually (so we can put the most important ones on top)

I recommend using something like https://github.com/bfirsh/django-ordered-model for this.

krysal commented:
Changes applied, I added descriptions and corrected names. I talked to Sarah for descriptions of some fields (e.g. Type of work) and she wants to rethink some headings for the cases, so probably tomorrow I will have to modify them.

kgodey commented on legal_db/models.py:
```diff
@@ -1,11 +1,135 @@
+        max_length=50, blank=True, null=True, help_text="The number of the case."
+    )
+    jurisdiction = models.CharField(
+        max_length=255, blank=True, null=True, help_text="Jurisdiction Within country."
```
Minor nitpick: can we make sure to keep the capitalization of help text consistent in the help text? I think we should use sentence case for help text.

kgodey replied:
I think this applies to other help text as well.

krysal replied:
Ups, yes, this was a little mistake. Thanks for spotting it. I agree in to keep text consistency.
